### PR TITLE
Replace Newtonsoft.Json with System.Text.Json

### PR DESCRIPTION
**Pull request by pkamphuis**
Replace Newtonsoft.Json with System.Text.Json with benefit of increased performance.

Still some clean up and validation needed

### CODE REVIEW

**Comment by jirikostiha** [comment]
Great. It is big change. Do you thing it will be well tested to not to break something?

**Comment by pkamphuis** [comment]
Testing is a tricky one.
I used xAPITest, which basically works correct. Apart from regular price calculation errors.
double Tp = 5392 * 1.1;
results 5931.200000000001
which is not accepted by the server. Double is not the best suitable type for prices. (decimal would be a better (i.e. exact) type)
However this error has nothing to do with the System.Text.Json conversion. I just verified the NewtonSoft version has the same issue.

I also tried creating a unit test project. However the code is not well suited for unit testing. For example I had to add a RestartCustomTag option to the CustomTag handler. To get consistent results.
However this works fine for running individual tests, but not when running multiple tests in parallel.

So this basically leaves the edge cases untested.

**Comment by pkamphuis** [comment]
I am reasonably convinced that it works. I verified an additional set of commands, including MargingLevel command for account balance info. CurrentUserData, ChartLast and TradingHours. All seems well.

File: src/SyncAPIConnector/responses/BaseResponse.cs [hunk 1]
```diff
@@ -1,66 +1,76 @@
...
             }
 
-            if (ob == null)
+            if (ob is null)
             {
                 throw new APIReplyParseException("JSON Parse exception: " + body);
             }
             else
             {
+#if DEBUG
```
Comment (pkamphuis) [inline src/SyncAPIConnector/responses/BaseResponse.cs]: Please verify is this is acceptable. It simplifies validating the json response content. But Console.Write should probably not be in the library code

File: src/SyncAPIConnector/responses/BaseResponse.cs [hunk 2]
```diff
@@ -1,66 +1,76 @@
...
+                {
+                    errCode = new ERR_CODE(ob["errorCode"]?.GetValue<string>());
+                    errorDescr = ob["errorDescr"]?.GetValue<string>();
+                }
+                this.customTag = ob["customTag"]?.GetValue<string>();
 
-                if (this.status == null)
+                if (this.status is null)
                 {
                     Console.Error.WriteLine(body);
```
Comment (jirikostiha) [inline src/SyncAPIConnector/responses/BaseResponse.cs]: Remove this Console also.

**Review by jirikostiha — CHANGES_REQUESTED** [review]
It seems good. Just remove Console lines.

File: src/SyncAPIConnector/responses/BaseResponse.cs (hunk 1 above)
Comment (jirikostiha) [reply src/SyncAPIConnector/responses/BaseResponse.cs]: Yes, should not be there. Pls remove it.

**Comment by jirikostiha** [comment]
> Testing is a tricky one. I used xAPITest, which basically works correct. Apart from regular price calculation errors. double Tp = 5392 * 1.1; results 5931.200000000001 which is not accepted by the server. Double is not the best suitable type for prices. (decimal would be a better (i.e. exact) type) However this error has nothing to do with the System.Text.Json conversion. I just verified the NewtonSoft version has the same issue.
> 
> I also tried creating a unit test project. However the code is not well suited for unit testing. For example I had to add a RestartCustomTag option to the CustomTag handler. To get consistent results. However this works fine for running individual tests, but not when running multiple tests in parallel.
> 
> So this basically leaves the edge cases untested.

I agree with you. It is actually old code and it is hard to test it. But with some changes we can improve at least something. 
Regarding doubles we have to deal with server api as it is. I tried to search for newer version or some alternatives in .net and found nothing. In my opinion xtb has to have newer api version for xstation because they do updates time to time, but it is probably not public. Also I did not get answers from them.

File: src/SyncAPIConnector/responses/BaseResponse.cs (hunk 1 above)
Comment (pkamphuis) [reply src/SyncAPIConnector/responses/BaseResponse.cs]: removed

**Review by jirikostiha — APPROVED** [review]
